types/estree/index.d.ts

```diff
@@ -104,6 +104,10 @@ export interface BlockStatement extends BaseStatement {
   innerComments?: Array<Comment> | undefined;
 }
 
+export interface StaticBlock extends Omit<BlockStatement, 'type'> {
```

Comment:
As it described here:
https://github.com/estree/estree/blob/master/es2022.md#staticblock

Reply:
Alternatively, you could introduce a `BaseBlock` type, eliminating the need for `Omit`.